openmls/src/group/core_group/kat_passive_client.rs

```diff
@@ -141,35 +140,17 @@ struct PassiveClient {
 }
 
 impl PassiveClient {
-    fn new(
-        group_config: MlsGroupConfig,
-        psk_ids: Vec<PreSharedKeyId>,
-        psks: Vec<ExternalPskTest>,
-    ) -> Self {
+    fn new(group_config: MlsGroupConfig, psks: Vec<ExternalPskTest>) -> Self {
         let backend = OpenMlsRustCrypto::default();
 
         // Load all PSKs into key store.
-        for psk_id in psk_ids.into_iter() {
-            let psk_bundle = {
-                let secret = {
-                    let entry = psks
-                        .iter()
-                        .find(|s| s.psk_id == psk_id.external_id().unwrap())
-                        .unwrap();
-
-                    Secret::from_slice(
-                        &entry.psk,
-                        ProtocolVersion::default(),
-                        group_config.crypto_config.ciphersuite,
-                    )
-                };
-
-                PskBundle::new(secret).unwrap()
-            };
-
-            backend
-                .key_store()
-                .store(&psk_id.tls_serialize_detached().unwrap(), &psk_bundle)
+        for psk in psks.into_iter() {
+            // TODO: Better API?
+            // We only construct this to easily save the PSK in the keystore.
```

Comment:
Feedback welcome.

Reply:
The only thin here I dislike is that we need to provide an empty nonce. What we really do need here, is to serialize the `Psk`, which is the `PskId` without the nonce. Maybe we shold operate on the `Psk` struct for this type of API instead.
We can do (something like) this in a follow up as well.